lib/base-class.js

```diff
@@ -0,0 +1,111 @@
+
+'use strct'
+
+const events = require('events')
+const os = require('os')
+const { spawn } = require('child_process')
+const path = require('path')
+const joinTrace = require('node-trace-log-join')
+const getLoggingPaths = require('./get-logging-paths')
+
+class ClinicCommon extends events.EventEmitter {
+  constructor (tool) {
+    super()
+    this.tool = tool
+  }
+
+  collect (args, callback) {
+    // Run program, but inject the sampler.
+    const logArgs = [
+      '-r', 'no-cluster.js',
+      '--trace-events-enabled', '--trace-event-categories'
+    ]
+
+    if (this.tool === 'bubbleprof') {
+      logArgs.unshift('-r', 'sampler.js')
+      logArgs.push('node.async_hooks')
+    } else if (this.tool === 'doctor') {
+      logArgs.unshift('-r', 'logger.js')
+      logArgs.push('v8')
+    }
```

Comment:
I would prefer a polymorphic approach (calling an abstract method) for these bits. So that doctor and bubbleprof could add their parameters